### DISCUSSION TITLE
Link component not working as expected

### DISCUSSION THREAD

**Discussion by SahilAggarwal2004** [serwist/serwist#47] [Q&A]
Continuing the discussion https://github.com/DuCanhGH/next-pwa/discussions/121#discussioncomment-8157267

I have actually precached the pages I am trying to refer to. After a client goes offline, the `Link` component doesn't work just for the first time (irrespective of the href/path), then onwards it works perfectly (which means that the pages are correctly cached).
```
Uncaught (in promise) TypeError: Failed to fetch
    at ec.fetch (sw.js:1:14808)
    at eC._handleFetch (sw.js:1:22712)
    at eC._handle (sw.js:1:22562)
    at async eC._getResponse (sw.js:1:17792)
fetch @ sw.js:1
_handleFetch @ sw.js:1
_handle @ sw.js:1
```
This is the error that is being logged when the `Link` component is clicked the first time

**Comment by SahilAggarwal2004** [comment]
Also, this issue started occuring when I migrated to app router from pages router. It didn't use to occur in pages router

**Comment by SahilAggarwal2004** [comment]
So sorry, I didn't see [this](https://github.com/serwist/serwist/discussions/46) before opening this discussion. No need to focus on this until you feel good...
Please take care of yourself, I hope you get well soon!
And Thanks for your awesome work till now!

**Comment by DuCanhGH** [comment]
@SahilAggarwal2004 that's... fun. The error seems too vague for me to tell what's going on, but it seems that the service worker  successfully intercepted the fetch yet wasn't able to fetch the page...? Not sure why this happens though.

**Comment by SahilAggarwal2004** [comment]
Actually, there's no big deal. This error occurs only if I switch to offline mode via Dev Tools. Evenything works as expected when the device is offline
